spec.html

```diff
@@ -1585,158 +1585,121 @@ <h1>Number::bitwiseNOT ( _x_ )</h1>
+          <emu-alg>
+            1. If _x_ is *NaN* or _y_ is *NaN*, return *NaN*.
+            1. If _x_ is *+&infin;*<sub>𝔽</sub> or _x_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _y_ is *+&infin;*<sub>𝔽</sub> or _y_ is *-&infin;*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is positive, return _x_.
+              1. If _x_ is *+&infin;*<sub>𝔽</sub>, return *-&infin;*<sub>𝔽</sub>. Otherwise, return *+&infin;*<sub>𝔽</sub>.
+            1. If _y_ is *+&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *+0*<sub>𝔽</sub>. Otherwise, return *-0*<sub>𝔽</sub>.
+            1. If _y_ is *-&infin;*<sub>𝔽</sub>, then
+              1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is positive, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
+            1. If _x_ is *+0*<sub>𝔽</sub> or _x_ is *-0*<sub>𝔽</sub>, then
+              1. If _y_ is *+0*<sub>𝔽</sub> or _y_ is *-0*<sub>𝔽</sub>, return *NaN*.
+              1. If _y_ is positive, return _x_.
+              1. If _x_ is *+0*<sub>𝔽</sub>, return *-0*<sub>𝔽</sub>. Otherwise, return *+0*<sub>𝔽</sub>.
+            1. If _y_ is *+0*<sub>𝔽</sub>, then
+              1. If _x_ is positive, return *+&infin;*<sub>𝔽</sub>. Otherwise, return *-&infin;*<sub>𝔽</sub>.
+            1. If _y_ is *-0*<sub>𝔽</sub>, then
+              1. If _x_ is positive, return *-&infin;*<sub>𝔽</sub>. Otherwise, return *+&infin;*<sub>𝔽</sub>.
+            1. Return 𝔽(ℝ(_x_) / ℝ(_y_)).
+          </emu-alg>
         </emu-clause>
 
         <emu-clause id="sec-numeric-types-number-remainder" oldids="sec-applying-the-mod-operator">
           <h1>Number::remainder ( _n_, _d_ )</h1>
-          <p>The abstract operation Number::remainder takes arguments _n_ (a Number) and _d_ (a Number). It yields the remainder of its operands from an implied division; _n_ is the dividend and _d_ is the divisor.</p>
+          <p>The abstract operation Number::remainder takes arguments _n_ (a Number) and _d_ (a Number). It yields the remainder from an implied division of its operands where _n_ is the dividend and _d_ is the divisor. It performs the following steps when called:</p>
+          <emu-alg>
+            1. If _n_ is *NaN* or _d_ is *NaN*, return *NaN*.
+            1. If _n_ is *+&infin;*<sub>𝔽</sub> or _n_ is *-&infin;*<sub>𝔽</sub>, return *NaN*.
+            1. If _d_ is *+0*<sub>𝔽</sub> or _d_ is *-0*<sub>𝔽</sub>, return *NaN*.
+            1. If _d_ is *+&infin;*<sub>𝔽</sub> or _d_ is *-&infin;*<sub>𝔽</sub>, return _n_.
+            1. If _n_ is *+0*<sub>𝔽</sub> or _n_ is *-0*<sub>𝔽</sub>, return _n_.
+            1. Let _r_ be ℝ(_n_) - (ℝ(_d_) &times; _q_) where _q_ is an integer that is negative if and only if _n_ and _d_ have opposite sign, and whose magnitude is as large as possible without exceeding the magnitude of ℝ(_n_) / ℝ(_d_).
```

Comment:
It is maybe worth asserting that `_n_` and `_d_` are finite and nonzero before this line.

The math looks right.

Reply:
I don't think it helps much, but I added it.